test/integration/propControls/index.spec.ts

```diff
@@ -0,0 +1,74 @@
+import { test, expect, Page, Locator } from '@playwright/test';
+import createApp from '../../utils/createApp';
+import clickCenter from '../../utils/clickCenter';
+import generateId from '../../utils/generateId';
+import { canvasFrame, selectedNodeEditor, pageRoot } from '../../utils/locators';
+import domInput from './domInput.json';
+
+async function getPropControlInputLocator(page: Page, inputPropName: string) {
+  const selectedNodeEditorLocator = page.locator(selectedNodeEditor);
+
+  const propControlLabelHandle = await selectedNodeEditorLocator
+    .locator(`label:has-text("${inputPropName}")`)
+    .elementHandle();
+  const propControlLabelFor = await propControlLabelHandle?.getAttribute('for');
+
+  return selectedNodeEditorLocator.locator(`input[id="${propControlLabelFor}"]`);
+}
+
+async function getInputElementLabelLocator(page: Page, inputLocator: Locator) {
+  const canvasFrameLocator = page.frameLocator(canvasFrame);
+
+  const inputHandle = await inputLocator.elementHandle();
```

Comment:
One potential way of avoiding relying on these implementation details is to add `Typography` elements in the same row and [layout](https://playwright.dev/docs/selectors#selecting-elements-based-on-layout) selectors to find the associated input, regardless of the label

Reply:
i can do that, i though using a label's `for` and the input `id` was actually better

Reply:
this is using accessibility props that should always match so i think it's a good approach (nothing implementation specific imo), we can change it if you insist though

Reply:
but the label should always match the id, right? that's how html inputs should work.

Reply:
or if it's an issue that we are kind of assuming the MUI inputs and labels have accessibility properly set up we can do it in another way - but it feels safe to assume that those accessibility props work.

Reply:
Yeah, I was thinking about the edge-cases
* can an input have an `aria-label`?
* if the `id` changes over the lifetime of the page, this locator will stop working as it only considers the id of the first time it executed

but that may be too far fetched

Reply:
i'll give locators another quick try, it's a more readable approach at least - for some reason i had some issues when i first tried due to the iframe, but now i understand things a bit better so maybe it's easy

Reply:
Yes, I haven't fully internalized them yet. The way I understand them is that they are like `document.queryAll`, only it's sort of a live collection. So if the underlying DOM changes, the content of the locators change with it. I feel like I haven't fully harnassed their power yet, it seems to sort of require you to think differently about selectors.

Reply:
layout selectors aren't working, maybe it's because the label intersects with the input, they don't seem very predictable at all... let's probably keep this approach for now, i don't think it should ever cause any issues.